packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -333,12 +336,62 @@ abstract class IntelliJValidator extends DoctorValidator {
 
       return;
     }
+    messages.add(new ValidationMessage.error(
+      '$title plugin not installed; this adds $title specific functionality.'
+    ));
+  }
+
+  Future<Null> _validateFlutterPackage(List<ValidationMessage> messages, List<String> packageNames, String title, {
+    Version minVersion
+  }) async {
 
+    for (String packageName in packageNames) {
+      if (!hasPackage(packageName)) {
+        continue;
+      }
+      final String versionText = _readPackageVersion(packageName);
+      final Version localVersion = new Version.parse(versionText);
+      if (localVersion != null && minVersion != null && localVersion < minVersion) {
+        messages.add(new ValidationMessage.error(
+          '$title plugin version $versionText - the recommended minimum version is $minVersion'
+        ));
+      } 
+      else {
+        
+        try {
+          final String latestFlutter = await _getLatestVersionPossible();
+          final Version flutterVersion = new Version.parse(latestFlutter);
+          final String serverMessage = localVersion < flutterVersion ? '[Consider upgrading your plugin version to $latestFlutter]' : '';
+            messages.add(new ValidationMessage(
+              '$title plugin ${localVersion != null ? "version $localVersion" : "installed"} $serverMessage'
+            ));
+        } catch (e) {          
```

Comment:
With the suggested change below of making the method return `null` rather than throwing a string, you should be able to add a `null` check on the latestVersion and remove this `catch` statement.